app/graphql/mushroom_observer_schema.rb

```diff
@@ -0,0 +1,43 @@
+# frozen_string_literal: true
+
+# NOTE: use `rails graphql:dump_schema`` to get a dump of the current schema
+
+class MushroomObserverSchema < GraphQL::Schema
+  query(Types::QueryType)
+  mutation(Types::MutationType)
+  # subscription(Types::Subscription)
+
+  default_max_page_size 24
+  # https://github.com/bibendi/graphql-connections
+  # https://relay.dev/graphql/connections.htm
+  # connections.add(ActiveRecord::Relation, GraphQL::Connections::Stable)
+
+  # GraphQL::Batch setup:
+  use GraphQL::Batch
+
+  # Union and Interface Resolution
+  def self.resolve_type(_abstract_type, _obj, _ctx)
+    # NOTE: Implement this function to return the correct object type for `obj`
+    # raise(GraphQL::RequiredImplementationMissingError)
+  end
+
+  # Relay-style Object Identification:
+
+  # Return a string UUID for `object`
+  def self.id_from_object(object, type_definition, query_ctx)
+    # Here's a simple implementation which:
```

Comment:
Just plain old object.id is guaranteed to be "unique within type", no reason to encode it.

Reply:
This is part of a group of necessary hooks stubbed out by the graphql-ruby gem author as examples - I think it's typically used for something specific to graphql, to do with giving "nodes" unique IDs, even when there are possible duplicate nested records, or ensuring some other "uniquity" within the query object, maybe because different records are mixed in the query. I commented out because I don't understand it yet!

I should just comment out the whole method, but the docs say it's necessary. They're not implemented anywhere yet as far as I can tell. Here's a partial [explanation](https://graphql-ruby.org/schema/definition#object-identification-hooks) from Robert Mosolgo.

Reply:
For example within the nodes of a query there could be multiple instances of the same user. I believe graphql wants each node to have a UUID (or already automatically gives them one).

```
observation {
  name
  user
  location
  species_list {
    user
  }
  sequence {
    user
  }
}